src/plugins/intel_cpu/src/nodes/mkldnn_fullyconnected_node.cpp

```diff
@@ -228,35 +228,42 @@ void MKLDNNFullyConnectedNode::prepareParams() {
     }
 
     appendPostOpArgs(*attr, primArgs, binaryPostOpsArgs);
+
+    auto reshapeMemory = [this](int argType) {
+        auto param = primArgs.find(argType);
+        if (param != primArgs.end()) {
+            auto oldMem = param->second;
+            auto dims = oldMem.get_desc().dims();
+            if (dims.size() == 3) {
+                std::vector<dnnl::memory::dim> normalizedDims({dims[0] * dims[1], dims[2]});
+                mkldnn::memory::desc newMemDesc(oldMem.get_desc().reshape(normalizedDims));
+                mkldnn::memory newMem(newMemDesc, oldMem.get_engine(), oldMem.get_data_handle());
+                primArgs.at(argType) = newMem;
+            }
+        }
+    };
+    reshapeMemory(DNNL_ARG_SRC);
+    reshapeMemory(DNNL_ARG_DST);
 }
 
 void MKLDNNFullyConnectedNode::execute(mkldnn::stream strm) {
     if (prim) {
-        auto reshapeMemory = [this](int argType) {
+        // in cases parameter -> FullyConnected or dynamic shapes
+        // we keep old pointer to data in primArgs on second iteration with same input shapes
+        auto updateMemoryPtr = [this](int argType) {
             auto param = primArgs.find(argType);
             if (param != primArgs.end()) {
-                auto oldMem = param->second;
-                auto dims = oldMem.get_desc().dims();
-                if (dims.size() == 3) {
-                    std::vector<dnnl::memory::dim> normalizedDims({dims[0] * dims[1], dims[2]});
-                    mkldnn::memory::desc newMemDesc(oldMem.get_desc().reshape(normalizedDims));
-                    mkldnn::memory newMem(newMemDesc, oldMem.get_engine(), oldMem.get_data_handle());
-                    primArgs.at(argType) = newMem;
-                } else {
-                    // in cases parameter -> FullyConnected or dynamic shapes
-                    // we keep old pointer to data in primArgs on second iteration with same input shapes
-                    if (argType == DNNL_ARG_SRC && getInputShapeAtPort(DATA_ID).getRank() == 3) {
-                        primArgs.at(argType).set_data_handle(getParentEdgesAtPort(0)[0]->getMemoryPtr()->GetData());
-                    }
-                    if (argType == DNNL_ARG_DST && getOutputShapeAtPort(0).getRank() == 3) {
-                        primArgs.at(argType).set_data_handle(getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetData());
-                    }
+                if (argType == DNNL_ARG_SRC && getInputShapeAtPort(DATA_ID).getRank() == 3) {
+                    primArgs.at(argType).set_data_handle(getParentEdgesAtPort(0)[0]->getMemoryPtr()->GetData());
+                }
+                if (argType == DNNL_ARG_DST && getOutputShapeAtPort(0).getRank() == 3) {
+                    primArgs.at(argType).set_data_handle(getChildEdgesAtPort(0)[0]->getMemoryPtr()->GetData());
```

Comment:
Do the SL tests check this behavior? I think probably not, since we need repeated shapes in the target shapes list. Am I right?

Reply:
Test case for static: https://github.com/openvinotoolkit/openvino/blob/16b260bee589c31244bf0e61cfd9aa94dd870e12/src/tests/functional/plugin/cpu/single_layer_tests/mat_mul.cpp#L358
Test case for dynamic: https://github.com/openvinotoolkit/openvino/blob/16b260bee589c31244bf0e61cfd9aa94dd870e12/src/tests/functional/plugin/cpu/single_layer_tests/mat_mul.cpp#L405

Reply:
Oh, I see. Thanks.